src/logcollector/logcollector.c

```diff
@@ -2611,17 +2598,17 @@ static void check_pattern_expand_excluded() {
 
 static void set_can_read(int value){
 
-    w_rwlock_wrlock(&can_read_rwlock);
-    _can_read = value;
-    w_rwlock_unlock(&can_read_rwlock);
+    RWLOCK_LOCK_WRITE(&can_read_rwlock, {
+        _can_read = value;
+    });
 }
 
 int can_read() {
 
     int ret;
```

Comment:
Is it necessary to declare this variable, because it can return `_can_read` directly?